src/cli.rs

```diff
@@ -647,26 +647,17 @@ pub struct Opts {
     )]
     search_path: Vec<PathBuf>,
 
-    /// Print results as YAML objects so you can use it with tools like yq and nushell.
+    /// Print results in a certain format so you can pipe it to tools.
     #[arg(
         long,
-        value_name = "yaml",
-        conflicts_with("format"),
-        conflicts_with("list_details"),
-        help = "Print results as YAML objects so you can use it with tools like yq and nushell."
-    )]
-    pub yaml: bool,
-
-    /// Print results as a JSON array so you can use it with tools like jq and nushell.
-    #[arg(
-        long,
-        value_name = "json",
+        value_name = "output",
```

Comment:
I'm still on the fence on this. `--output` allows a more general framework for output formats, but `--json` or `--jsonl` is easier to use. And `ripgrep` uses `--json` (but outputs jsonl).

I'm also not totally convinced that we need more than just jsonl. What is the use case for using yaml or a full json array? I suppose yaml is probably more human readable.  Maybe a tabular format could make sense at some point?

I'm curious what @sharkdp's thoughts are.

Reply:
In my opinion, JSONL, as an extension to JSON format, naturally gets much fewer support than JSON and YAML.

For example, I googled some time and found out that I have to use `jq --stream` , not `--jsonl` or `--ndjson` to parse a JSONL stdin. As another example, the `nushell` supports JSON, YAML and TOML out-of-box, but can only work with JSONL with an extension. 

Looking forward to sharkdp's thoughts.

Reply:
As another data point here, ripgrep's `--json` flag emits JSONL. It has been doing so for years and I don't think I've ever heard anyone complain that was using JSONL instead of just JSON. I indeed chose JSONL because it made streaming the output essentially trivial.